Source/Csla.AspNetCore/ApplicationContextManager.cs

```diff
@@ -3,38 +3,53 @@
 //     Copyright (c) Marimer LLC. All rights reserved.
 //     Website: https://cslanet.com
 // </copyright>
-// <summary>Application context manager that uses HttpContextAccessor</summary>
+// <summary>Application context manager that adapts for AspNetCore/SSB</summary>
 //-----------------------------------------------------------------------
 using Csla.Core;
+using System;
+#if NET5_0_OR_GREATER
+using Csla.AspNetCore.Blazor;
+using Microsoft.Extensions.DependencyInjection;
+#else
 using Microsoft.AspNetCore.Http;
-using System.Security.Claims;
+#endif
 
 namespace Csla.AspNetCore
 {
   /// <summary>
-  /// Application context manager that uses HttpContextAccessor when 
-  /// resolving HttpContext to store context values.
+  /// Application context manager that adapts for AspNetCore/SSB.
   /// </summary>
-  public class ApplicationContextManager : IContextManager
+  public class ApplicationContextManager : IContextManager, IDisposable
   {
+    private IContextManager ActiveContextManager { get; set; }
 
-    private const string _localContextName = "Csla.LocalContext";
-    private const string _clientContextName = "Csla.ClientContext";
-
+#if NET5_0_OR_GREATER
     /// <summary>
-    /// Creates an instance of the object, initializing it
-    /// with the required IServiceProvider.
+    /// Creates an instance of the object.
     /// </summary>
-    /// <param name="httpContextAccessor">HttpContext accessor</param>
-    public ApplicationContextManager(IHttpContextAccessor httpContextAccessor)
+    /// <param name="provider"></param>
+    /// <param name="activeCircuitHandler"></param>
+    public ApplicationContextManager(IServiceProvider provider, ActiveCircuitHandler activeCircuitHandler)
```

Comment:
I'm not sure this is a great idea, but I'll mention it anyway ...!

I think there might be an opportunity to avoid the need to duplicate the Blazor context manager by not depending on the circuit handler here. If we instead depended on a separate class that the circuit handler is able to influence, the context manager itself is independent of server-side Blazor. That change might allow the circuit handler to live in Csla.AspNetCore, but the hybrid context manager to live in Csla.Blazor.

The only problem with this is that the type on which this class depends would have to live somewhere that is available in both flavours of Blazor, but also accessible from Csla.AspNetCore. Does that mean it would have to go in the core Csla assembly? It doesn't really feel right for such a class to go in the root Csla assembly.

Reply:
I struggled with that, and didn't want to add such a type to core Csla.dll. 

There is no duplicate Blazor context manager now though, because Csla.Blazor no longer contains a context manager.

There is a "duplicate" in that Csla.Blazor.WebAssembly now contains a context manager, because the aspnetcore one has access to that server-hosted authentication state provider interface, and the wasm one does not. That feels a little dirty to me, but maybe it is better to have SSB and wasm be cleanly separate in their implemenations.

Reply:
Rocky I humbly remember here what you told me over 10+ years ago in person at a MS conference in San Fran...when I was struggling with duplicated code in different objects.  I was trying to figure out how to share Property Getters between read-only and writeable objects.  You said two things:
1.  "Behavior driven design" and you had me go buy David Wests book :-)
2.  "If your not using code generation then your an idiot"  You said it nicely :-)

The application context manager that serves WASM is different from SSB according to that sage wisdom....only because Microsoft is doing different things under the hood with them (their version of ServerAuthenticationStateProvider)